src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala

```diff
@@ -82,8 +82,31 @@ class CloudStackVaultController(vaultService: VaultService,
     logger.debug(s"handleUserCreate(userId: $userId)")
 
     val accountId = cloudStackService.getAccountIdByUserId(userId)
+    val usersIds = cloudStackService.getUserIdsByAccountId(accountId)
+
+    val currentVaultTags = getCurrentUsersVaultTags(usersIds)
+
+    val newVaultTokenTagKeyList = Set(Tag.Key.VaultRO, Tag.Key.VaultRW).collect {
+      case tagKey if !currentVaultTags.exists(_.key == tagKey) => createMissingAccountTokenTag(accountId, tagKey)
+    }
+
+    val newVaultKeyspaceTagKeyList = Set(Tag.Key.VaultHost, Tag.Key.VaultPrefix).collect {
+      case tagKey if !currentVaultTags.exists(_.key == tagKey) =>
+        tagKey match {
+          case Tag.Key.VaultHost => Tag.createTag(Tag.Key.VaultHost, vaultApiPath)
+          case Tag.Key.VaultPrefix => Tag.createTag(Tag.Key.VaultPrefix, getAccountEntitySecretPath(accountId))
+        }
+    }
 
-    handleAccountCreate(accountId)
+    val completeVaultTags: Set[Tag] = currentVaultTags ++ newVaultTokenTagKeyList ++ newVaultKeyspaceTagKeyList
```

Comment:
convert "completeVaultTags" to List here  (before if)

Reply:
ok